File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/ITransactionApiContainer.cs
using System;
using JetBrains.Annotations;

namespace Volo.Abp.Uow;

public interface ITransactionApiContainer
{
    [CanBeNull]
    ITransactionApi FindTransactionApi(@NonNull String key);

    void AddTransactionApi(@NonNull String key, @NonNull ITransactionApi api);

    [NotNull]
    ITransactionApi GetOrAddTransactionApi(@NonNull String key, @NonNull Func<ITransactionApi> factory);
}

File: ueueo-frameworks/ueueo-uow/src/main/java/com/ueueo/uow/ITransactionApiContainer.java
package com.ueueo.uow;

import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import java.util.function.Supplier;

/**
 * @author Lee
 * @date 2022-05-19 20:46
 */
public interface ITransactionApiContainer {
    @Nullable
    ITransactionApi findTransactionApi(@NonNull String key);

    void addTransactionApi(@NonNull String key, @NonNull ITransactionApi api);

    @NonNull
    ITransactionApi getOrAddTransactionApi(@NonNull String key, @NonNull Supplier<ITransactionApi> factory);
}
